File: frontend/src/api/market.ts
import { api } from './client'
import type { MarketItem } from './types'

export interface MarketItemCreatePayload {
  name: string
  category?: string
}

export const marketApi = {
  list: () => api.get<MarketItem[]>('/market'),
  create: (payload: MarketItemCreatePayload) => api.post<MarketItem>('/market/items', payload),
  updateStatus: (id: number, status: MarketItem['status']) =>
    api.patch<MarketItem>(`/market/items/${id}`, { status }),
  remove: (id: number) => api.delete<void>(`/market/items/${id}`),
}

File: backend/api/schemas/market.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from db.models.enums import MarketItemStatus


class MarketItemCreate(BaseModel):
    name: str
    category: Optional[str] = None


class MarketItemStatusUpdate(BaseModel):
    """`API.md` documents this endpoint as `PATCH /market/items/{id} (status)` —
    a status transition, not a generic field editor. No other field of
    `market_items` has a documented post-creation edit path."""

    status: MarketItemStatus


class MarketItemOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    category: Optional[str]
    is_recurring_suggestion: bool
    status: MarketItemStatus
    created_at: datetime
